### ISSUE TITLE
MDM deprecated - replace by PDM?

### ISSUE THREAD

**Issue by Vertygo** [Luncher91/ActivDbgNET#4]
Hi @Luncher91,

I have a quick look at what you are building and it looks interesting to me. I have installed VS 2017 remote debugger and tried running ActivDbgNET but it fails on `Type.GetTypeFromProgID("MDM.AD1");` inside of `IMachineDebugManager GetMDM()` method.

The error indicates that you are trying to instantiate MDM that is deprecated? [This link](https://blogs.msdn.microsoft.com/greggm/2008/06/09/writing-a-script-debugger-without-using-the-mdm/) suggest using PDM instead.

Also, would this debugger also be able to debug ASP classic (eg attach to w3wp process)?

**Comment by Luncher91** [comment]
Hi,

I was able to compile the library using VS 2015 Enterprise.

You might be right. I hardly found anything about debugging interfaces to debug script hosts. So I tinkered around and found that way to work. Maybe using the Microsoft.VisualStudio.Debugger.Interop namespace is a more comfortable way of implementing a debugger. I think an evaluation on that would definitly be worth it.
Maybe we could get rid of ActivDbg.dll and use instead the Microsoft.VisualStudio.Debugger.Interop.dll which comes with VS Studio 14.0. That would definitly be a plus.

The main purpose is to debug scripts running in any script host. I don't know enough about the w3wp process to tell for sure. But I assume that w3wp uses the common script host and therefor can be debugged as well. I was testing it usually with cscript or wscript.

**Comment by Vertygo** [comment]
Hi,

Or I might be completely wrong ... after reading a little bit, [here](https://docs.microsoft.com/en-us/visualstudio/extensibility/debugger/debugger-components?view=vs-2017) I found out that PDM/SDM might be used just to provide debugging extension for Visual Studio. Or, it might be that Microsoft.VisualStudio.Debugger.Interop uses PDM/SDM to enable debugging for VS only. 

I have found an example of how to build Lua debugger extension for VS [here](https://github.com/Microsoft/VSLua/) using VisualStudio.Debugger package.
And I also found JS/VBScript debugger [here](https://github.com/Microsoft/ClearScript/) that **might be interesting**...

I have tried to do something with ClearScript example but I lack some knowledge with this kind of stuff, hope you might find it more useful. :)
